### PR TITLE
Frontend backend integration

### PR DESCRIPTION
**Pull request by MaximilianWagner**
Add file field to store firmware file.
change processing flow:
- file uploading independent of analysing.
- on analysing the firmware file gets unpacked copied to a new directory and processed by emba.sh
- able to select firmware to be processed before analysing

### CODE REVIEW

File: embark/uploader/models.py [hunk 1]
```diff
@@ -92,19 +116,22 @@ class Meta:
 
         :return:
     """
-    def to_shell(self):
-        path = f"{self.id}/{self.title}"
-        command = f"{path}"
+
+    def __init__(self, *args, **kwargs):
```
Comment (diegiesskanne) [inline embark/uploader/models.py:120]: can be removed (not used)

File: embark/uploader/views.py [hunk 2]
```diff
@@ -61,15 +63,25 @@ def upload_file(request):
...
+            firmware_flags = Firmware.objects.latest('id')
+
+            # inject into bounded Executor
+            if boundedExecutor.submit_firmware(firmware_flags=firmware_flags, firmware_file=firmware_file):
+                return HttpResponseRedirect("../../home/#uploader")
+            else:
+                return HttpResponse("queue full")
         else:
             logging.info("Posted Form is unvalid")
+            print(form.errors)
```
Comment (diegiesskanne) [inline embark/uploader/views.py:80]: debug?

File: embark/uploader/models.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
+from django.conf import settings
```
Comment (diegiesskanne) [inline embark/uploader/models.py:1]: a few comments in this file could be helpful

**Review by diegiesskanne — APPROVED** [review]
good stuff

File: embark/uploader/views.py (hunk 2 above)
Comment (ashutoshsingh0223) [reply embark/uploader/views.py:80]: Can be removed and replaced with logger entry

File: embark/uploader/archiver.py [hunk 4]
```diff
@@ -88,9 +87,39 @@ def unpack(file_location, extract_dir=None):
...
+            :param file_name: file to be checked
+
+            :return: True if extension is supported, ValueError otherwise
+        """
+
+        ext = ".{0}".format(file_name.rsplit(".", 1)[1])
+        if cls.get_supported_extensions().__contains__(ext):
+            return True
+
+        logging.error("Format .%s is not supported", ext)
```
Comment (ashutoshsingh0223) [inline embark/uploader/archiver.py:124]: use logger to log to file. That way these logs persist for a while to see.

File: embark/uploader/boundedExecutor.py [hunk 5]
```diff
@@ -42,12 +43,11 @@ def run_shell_cmd(self, cmd):
...
 
-            # TODO emba.log need to be informed
-            emba_process = subprocess.run(cmd, shell=True, check=True)
+            # TODO: progress bar needs to be started
+            emba_process = subprocess.call(cmd, shell=True)
 
             # success
-            logging.info("cmd run successful")
-            logging.info(emba_process.returncode)
+            logging.info("Success: {0}".format(cmd))
```
Comment (ashutoshsingh0223) [inline embark/uploader/boundedExecutor.py:50]: Use logger. Similar comment for all other logging instances

**Review by ashutoshsingh0223 — COMMENTED** [review]
Very nicely done.
1. Very nice usage of Base Field classes and modification to our use case
2. Good job with adding `TODO` everywhere with scope/requirement of future work
3. Added tests for functions

Very minor suggestions:
1. Use file logger: It may seem that is there is no difference but makes your app ready for integration with log lakes etc.
2. Does not crowd stdout and lifecycle is under your control. For eg. I can delete web.log without deleting docker logs.
3.  Python classes start with capital letter - `BoundedExecutor`. Don't know why linter is not picking it up.

**Review by ashutoshsingh0223 — APPROVED** [review]

**Pull request merged**